solr/core/src/java/org/apache/solr/cluster/placement/PropertyKey.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.cluster.placement;
+
+/**
+ * A property key used by plugins to request values from Solr
+ */
+public interface PropertyKey {
+  /**
+   * @return the target of this {@link PropertyKey}, i.e. from where the corresponding {@link PropertyValue}'s should be
+   * (or were) obtained.
+   */
+  PropertyValueSource getPropertyKeyTarget();
```

Comment:
Uh ... why not use `getPropertyValueSource` or `getTarget` ? mixing the "key", "value", "source" and "target" in one method is very confusing. Javadoc can explain what the method actually does.